microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/ReadinessChecker.java

```diff
@@ -0,0 +1,101 @@
+package org.jboss.eap.qe.microprofile.health.manual.delay;
+
+import static io.restassured.RestAssured.get;
+import static org.jboss.eap.qe.microprofile.health.manual.delay.ReadinessState.DOWN_NO_CHECK;
+import static org.jboss.eap.qe.microprofile.health.manual.delay.ReadinessState.DOWN_NO_CONTENT;
+import static org.jboss.eap.qe.microprofile.health.manual.delay.ReadinessState.UNABLE_TO_CONNECT;
+import static org.jboss.eap.qe.microprofile.health.manual.delay.ReadinessState.UP_NO_CHECK;
+import static org.jboss.eap.qe.microprofile.health.manual.delay.ReadinessState.UP_WITH_CHECK;
+import static org.jboss.eap.qe.microprofile.health.manual.delay.ReadinessState.UP_WITH_DEFAULT_CHECK;
+
+import java.net.ConnectException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.jboss.eap.qe.microprofile.health.delay.DelayedReadinessHealthCheck;
+import org.jboss.eap.qe.microprofile.health.tools.HealthUrlProvider;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+
+import io.restassured.internal.RestAssuredResponseImpl;
+import io.restassured.response.Response;
+
+/**
+ * Performs calls to the readiness health checks probes and stores the returned states
+ */
+public class ReadinessChecker implements Callable<Boolean> {
+
+    private AtomicBoolean shouldStop = new AtomicBoolean(false);
+
+    private List<ReadinessState> states = new CopyOnWriteArrayList<>();
+
+    public static final String DEFAULT_READINESS_CHECK_NAME_PREFIX = "ready-deployment.";
+
+    @Override
+    public Boolean call() {
+        states.add(ReadinessState.START());
```

Comment:
What about to use `addState` to be consistent in this method? Same for `states.add(ReadinessState.END());`

Reply:
Right, fixing.